File: backend/app/schemas/telegram_integration.py
from datetime import datetime

from pydantic import BaseModel, Field


class TelegramConnectInput(BaseModel):
    bot_token: str = Field(min_length=1)
    channel_id: str = Field(min_length=1)


class TelegramStatusOut(BaseModel):
    is_connected: bool
    channel_id: str | None
    created_at: datetime | None


class TelegramPublishInput(BaseModel):
    text: str = Field(min_length=1)


class TelegramPublishOut(BaseModel):
    message_id: int

File: frontend/app/types/telegramIntegration.ts
export interface TelegramStatus {
    is_connected: boolean
    channel_id: string | null
    created_at: string | null
}

export interface TelegramConnectInput {
    bot_token: string
    channel_id: string
}

export interface TelegramPublishOutput {
    message_id: number
}